### PR TITLE
ACMS-1242: Update UI installation to ask for questions

### PR DESCRIPTION
**Pull request by rajeshreeputra**

### CODE REVIEW

File: src/Form/SelectionWizardForm.php [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Render\Renderer;
+use Drupal\Core\State\StateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a Acquia Tour form.
+ */
+class SelectionWizardForm extends FormBase {
```
Comment (panshulK) [inline src/Form/SelectionWizardForm.php:17]: @rajeshreeputra , @vishalkhode1, @chandan-singh7929  Shouldn't we change the class file name to something that indicates the functionality implemented? Something more apt. like - QuestionWizardForm ?

File: src/Form/StarterKitSelectionForm.php [hunk 2]
```diff
@@ -140,8 +140,17 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     ];
     $form['tour-dashboard']['message'] = [
         '#type' => 'markup',
-        '#markup' => '<p>' . $this->t("Acquia CMS Starter kits can be used to install Acquia CMS as per the requirement i.e. with or without content model or Site Studio.
-        You can either select the starter kit now or do it later. Doing this would enable the required modules for the selected starter kit.") . '</p>',
+        '#markup' => '<p>' . $this->t("Acquia CMS starter kits provide different starting points for your site depending on your requirements. Select a from one of the starter kits below to enable the modules.") . '</p>',
```
Comment (panshulK) [inline src/Form/StarterKitSelectionForm.php:143]: Grammatical error - "Select a from". @rajeshreeputra 

Also, I think. we need to re-write this - "Acquia CMS starter kits provide different starting points for your site depending on your requirements. Select a from one of the starter kits below to enable the modules."

Suggestions @chandan-singh7929 @vishalkhode1 ?

**Comment by rajeshreeputra** [comment]
> @rajeshreeputra - On `admin/tour/dashboard` page I can see wizard option for selecting starterkit, but after selection I can also see a message that should only appear, if that are no module enable. we need to check that. image
> 
> Also, on wizard page I can see message like `Fields marked * are required' but I don't see any field required, do we need that message. image
> 
> Lastly, we should trigger wizard setup automatically once user done with setting up starter kit, this is not happening currently.


- Removed `Fields marked * are required' message
- Installation wizard set to trigger automatically after starter kit selection

File: src/Form/StarterKitSelectionForm.php [hunk 3]
```diff
@@ -153,20 +162,41 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ],
       '#default_value' => $this->state->get('acquia_cms.starter_kit'),
     ];
-    $form['tour-dashboard']['table'] = [
-      '#type' => 'table',
-      '#header' => $header,
-      '#rows' => $rows,
+    $form['tour-dashboard']['starter_kit_questions']['demo_question'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Do you want a demo with demo content (yes/no) ?'),
```
Comment (panshulK) [inline src/Form/StarterKitSelectionForm.php:167]: @vishalkhode1  @chandan-singh7929  @rajeshreeputra  need suggestion here as well. Language of the question can be changed IMO.

**Review by panshulK — CHANGES_REQUESTED** [review]

File: src/Form/SelectionWizardForm.php (hunk 1 above)
Comment (rajeshreeputra) [reply src/Form/SelectionWizardForm.php:17]: okay we can change the class file name to StarterKitSelectionWizardForm

**Comment by panshulK** [comment]
> > @rajeshreeputra - On `admin/tour/dashboard` page I can see wizard option for selecting starterkit, but after selection I can also see a message that should only appear, if that are no module enable. we need to check that. image
> > Also, on wizard page I can see message like `Fields marked * are required' but I don't see any field required, do we need that message. image
> > Lastly, we should trigger wizard setup automatically once user done with setting up starter kit, this is not happening currently.
> 
> * Removed `Fields marked * are required' message
> * Installation wizard set to trigger automatically after starter kit selection

The form (actual selections) can be center aligned. @rajeshreeputra

File: src/Form/StarterKitSelectionForm.php (hunk 2 above)
Comment (rajeshreeputra) [reply src/Form/StarterKitSelectionForm.php:143]: updated

File: js/acquia_cms_tour_dashboard.js [hunk 4]
```diff
@@ -10,12 +10,67 @@
...
     attach: function (context, settings) {
       $('.acms-dashboard-form-wrapper', context).once('acquiaCmsDashboardDialog').each(function () {
-        if (!settings.hide_starter_kit_wizard_modal && !settings.selected_starter_kit) {
+        if (!settings.hide_starter_kit_wizard_modal && !settings.selected_starter_kit && settings.show_starter_kit_modal) {
           $('.acms-starterkit-modal-form').click();
         }
         if (settings.show_wizard_modal && !settings.wizard_completed){
           $('.acms-dashboard-modal-form').click();
         }
+        var checkStarterKitExist = setInterval(function() {
```
Comment (vishalkhode1) [inline js/acquia_cms_tour_dashboard.js:19]: @rajeshreeputra Any reason for using setInterval. I think we should definitely not use it here  and same for below.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**